### PR TITLE
Allow unauthenticated access to the metrics endpoint.

### PR DESCRIPTION
**Pull request by tmcgilchrist**
The main change here is to align the auth check between GitHub and GitLab, in particular the `/metrics` endpoint used by prometheus was unnecessarily requiring authentication. 

Second I explicitly matched on variants in match statements, both to make it clearer what the roles are and for completeness.

### CODE REVIEW

File: gitlab/pipeline.ml [hunk 1]
```diff
@@ -4,6 +4,18 @@ module Git = Current_git
 module Gitlab = Current_gitlab
 module Docker = Current_docker.Default
 
+module Metrics = struct
+  open Prometheus
+
+  let namespace = "ocurrent"
```
Comment (maiste) [inline gitlab/pipeline.ml:10]:
```suggestion
  let namespace = "ocaml-ci"
```

File: gitlab/pipeline.ml [hunk 2]
```diff
@@ -136,6 +148,9 @@ let set_active_repos ~(installation : Installation.t Current.t)
        (fun acc r -> Index.Repo_set.add r.Gitlab.Repo_id.name acc)
        Index.Repo_set.empty
   |> Index.set_active_repos ~owner:installation.Installation.name;
+  Prometheus.Gauge.set
+    (Metrics.repositories_total installation.Installation.name)
+    (float_of_int (List.length repos));
```
Comment (maiste) [inline gitlab/pipeline.ml:153]: Shouldn't this setting system be hidden into the metrics, so we don't have to worry about the type? Like:
```ocaml
Metrics.set_repositories_total installation.Installation.name (float_of_int (List.length repos));
```
This way we have more control over the metrics inside the `Metrics` module. WDYT?

**Review by maiste — APPROVED** [review]
Two small questions but otherwise, LGMT :+1:

File: gitlab/pipeline.ml (hunk 1 above)
Comment (tmcgilchrist) [reply gitlab/pipeline.ml:10]: This aligns with the equivalent metrics in the GitHub plugin https://github.com/ocurrent/ocurrent/blob/master/plugins/github/installation.ml#L4-L13

In future the GitLab plugin will have a table of registered org/repos that can feed this Metric.

Comment (maiste) [reply gitlab/pipeline.ml:10]: Oh, is it because this is considered to be `current_gitlab`? :thinking:

Comment (tmcgilchrist) [reply gitlab/pipeline.ml:10]: Basically yes.

File: gitlab/pipeline.ml (hunk 2 above)
Comment (tmcgilchrist) [reply gitlab/pipeline.ml:153]: Sure, when this gets moved into the ocurrent plugin.

**Pull request merged**